### PR TITLE
Add secrets patch to buildah

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: umohnani8 <umohnani@redhat.com>
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

File: tests/secrets.bats [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    echo "$output"
+	[ "$status" -eq 0 ]
+	mounts="$output"
+    run grep "secrets" <<< "$mounts"
+    echo "$output"
+    [ "$status" -eq 0 ]
+	buildah rm $cid
+    rm -rf $TESTSDIR/containers
+    rm -rf $TESTSDIR/rhel
+}
```
Comment (nalind) [inline tests/secrets.bats]: The indentation in this shell function mixes tabs and spaces inconsistently.

File: secrets.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+		ctrDirOnHost := filepath.Join(containerWorkingDir, ctrDir)
+		if err = os.RemoveAll(ctrDirOnHost); err != nil {
+			return nil, fmt.Errorf("remove container directory failed: %v", err)
+		}
+
+		if err = os.MkdirAll(ctrDirOnHost, 0755); err != nil {
+			return nil, fmt.Errorf("making container directory failed: %v", err)
+		}
+
+		hostDir, err = resolveSymbolicLink(hostDir)
```
Comment (nalind) [inline secrets.go]: Is this step necessary?

**Comment by nalind** [comment]
A couple of nits, but otherwise it looks reasonable.

File: secrets.go (hunk 2 above)
Comment (rhatdan) [reply secrets.go]: I would guess this came from the original patch used for docker.

File: tests/secrets.bats (hunk 1 above)
Comment (rhatdan) [reply tests/secrets.bats]: Fixed

**Comment by rhatdan** [comment]
Will merge when the tests pass.  And then will build new version of buildah

**Pull request merged**